default/python/AI/MilitaryAI.py

```diff
@@ -104,32 +104,19 @@ def get_military_fleets(mil_fleets_ids=None, try_reset=True, thisround="Main"):
     global _military_allocations, totMilRating, num_milships
 
     universe = fo.getUniverse()
-    empire = fo.getEmpire()
-    empire_id = empire.empireID
-    capital_id = PlanetUtilsAI.get_capital()
-    if capital_id is None:
-        homeworld = None
-    else:
-        homeworld = universe.getPlanet(capital_id)
-    if homeworld:
-        home_system_id = homeworld.systemID
-    else:
-        home_system_id = -1
+    empire_id = fo.empireID()
+    home_system_id = PlanetUtilsAI.get_capital_sys_id()
+
+    all_military_fleet_ids = (mil_fleets_ids if mil_fleets_ids is not None
+                              else FleetUtilsAI.get_empire_fleet_ids_by_role(MissionType.MILITARY))
 
-    if mil_fleets_ids is not None:
-        all_military_fleet_ids = mil_fleets_ids
-    else:
-        all_military_fleet_ids = FleetUtilsAI.get_empire_fleet_ids_by_role(MissionType.MILITARY)
     if try_reset and (fo.currentTurn() + empire_id) % 30 == 0 and thisround == "Main":
         try_again(all_military_fleet_ids, try_reset=False, thisround=thisround + " Reset")
 
-    num_milships = 0
-    for fid in all_military_fleet_ids:
-        num_milships += foAI.foAIstate.fleetStatus.get(fid, {}).get('nships', 0)
+    num_milships = sum((foAI.foAIstate.fleetStatus.get(fid, {}).get('nships', 0) for fid in all_military_fleet_ids))
```

Comment:
If you pass generator expression to function you can omit parentheses.
`sum((a for a in x))` -> `sum(a for a in x)`